fastpay/src/server.rs

```diff
@@ -204,27 +278,76 @@ fn main() {
             rt.block_on(join_all(handles));
         }
 
-        ServerCommands::Generate {
-            protocol,
-            host,
-            port,
-            shards,
-        } => {
-            let key = KeyPair::generate();
-            let name = key.public();
-            let authority = AuthorityConfig {
-                network_protocol: protocol,
-                name,
-                host,
-                base_port: port,
-                num_shards: shards,
-            };
-            let server = AuthorityServerConfig { authority, key };
+        ServerCommands::Generate { options } => {
+            let path = options.server_config_path.clone();
+            let server = make_server_config(options);
             server
-                .write(server_config_path)
+                .write(&path)
                 .expect("Unable to write server config file");
             info!("Wrote server config file");
             server.authority.print();
         }
+
+        ServerCommands::GenerateAll {
+            authorities,
+            committee,
+        } => {
+            let mut rng = coconut::rand::thread_rng();
+            let parameters = coconut::Parameters::new(
+                3,
+                /* TODO: check party capacity for bulletproofs */ authorities.len(),
```

Comment:
@asonnino Is this correct?

Reply:
No, this is the max number of output coins that can be produced. We use the aggregation feature of bulletproofs to pay log cost in the number of range proofs.